src/Apps/Legion/Components/DataPusher/DataPusher.tsx

```diff
@@ -40,21 +40,30 @@ const DataPusher: React.FC<IDataPusherProps> = (props) => {
         theme
     });
 
+    //side-effects
+    useEffect(() => {
+        setSelectedClusterUrl(adapter.connectionString);
+    }, [adapter]); // keep the dependency list as is
+
     logDebugConsole('debug', 'Render');
     return (
         <DataPusherContext.Provider
             value={{
                 adapter,
+                selectedClusterUrl,
                 classNames
             }}
         >
             <div className={classNames.root}>
                 <h3>{t('legionApp.dataPusher.title')}</h3>
                 <ClusterPicker
-                    onClusterUrlChange={setSelectedClusterUrl}
+                    onClusterUrlChange={(clusterUrl: string) =>
```

Comment:
why do we need the wrapper function now? Can't we just pass the function directly if the props are the same?